src/main/java/com/rackspace/salus/monitor_management/services/TestMonitorService.java

```diff
@@ -111,18 +114,28 @@ public TestMonitorService(MonitorConversionService monitorConversionService,
         .setTenantId(tenantId)
         .setResourceId(resourceId);
 
-    final Resource resource = resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
-        .orElseThrow(() -> new MissingRequirementException("Unable to locate the resource for the test-monitor"));
+    final Optional<Resource> optionalResource = resourceRepository
+        .findByTenantIdAndResourceId(tenantId, resourceId);
+    if (optionalResource.isEmpty()) {
+      return CompletableFuture.completedFuture(new TestMonitorResult()
+          .setErrors(List.of("Unable to locate the resource for the test-monitor")));
+    }
+
+    Resource resource = optionalResource.get();
 
     final String envoyId;
-    if (isRemote) {
-      final RemoteMonitorDetails remoteMonitorDetails = (RemoteMonitorDetails) details;
-      List<String> monitoringZones = monitorManagement.determineMonitoringZones(
-          remoteMonitorDetails.getMonitoringZones(),
-          resource.getMetadata().get(REGION_METADATA));
-      envoyId = resolveRemoteEnvoy(tenantId, monitoringZones);
-    } else {
-      envoyId = resolveLocalEnvoy(tenantId, resourceId);
+    try {
+      if (isRemote) {
+        final RemoteMonitorDetails remoteMonitorDetails = (RemoteMonitorDetails) details;
+        List<String> monitoringZones = monitorManagement.determineMonitoringZones(
+            remoteMonitorDetails.getMonitoringZones(),
+            resource.getMetadata().get(REGION_METADATA));
+        envoyId = resolveRemoteEnvoy(tenantId, monitoringZones);
+      } else {
+        envoyId = resolveLocalEnvoy(tenantId, resourceId);
+      }
+    } catch (MissingRequirementException e) {
```

Comment:
Could also catch IllegalArgumentException to accomodate

https://github.com/racker/salus-telemetry-monitor-management/pull/190/files?diff=unified&w=1#diff-4de60ab5384dc92a33667f8259740fd2R191

```suggestion
    } catch (MissingRequirementException|IllegalArgumentException e) {
```